infra/terraform/modules/kubernetes_cluster/kops.tf

```diff
@@ -0,0 +1,173 @@
+resource "null_resource" "check_supported_k8s_version" {
+  triggers {
+    kubernetes_version = "${var.kubernetes_version}"
+    kubernetes_versions = "${join(",", sort(var.supported_k8s_versions))}"
+  }
+  provisioner "local-exec" {
+    command = <<EOF
+status=${contains(var.supported_k8s_versions, var.kubernetes_version) ? 0 : 1}
+if [[ $status == "1" ]]
+then
+  echo "Kubernetes version ${var.kubernetes_version} not supported"
+fi
+exit $status
+EOF
+  }
+}
+
+resource "null_resource" "check_kops_version" {
+  depends_on = [
+    "null_resource.check_supported_k8s_version"
+  ]
+
+  provisioner "local-exec" {
+    command = "${path.module}/bin/check_kops_version.sh ${path.module}/data/user_data.sh"
```

Comment:
I may be just tired so sorry if it's a stupid question: What is this command exactly doing? Is it normal you're calling two scripts here? Is there a `&&` missing or maybe it's implicit?

Reply:
Yeah that's not enormously clear, and it's a bit weird. `check_kops_version.sh` takes the path to the `user_data.sh` script as an argument, then greps it for the version of `kops` that's going to be installed on cluster instances, and checks it's the same as the version of kops installed locally.